app/queries/articles/active_threads_query.rb

```diff
@@ -0,0 +1,32 @@
+module Articles
+  class ActiveThreadsQuery
+    DEFAULT_OPTIONS = {
+      tags: ["discuss"],
+      time_ago: nil,
+      count: 10
+    }.with_indifferent_access.freeze
+
+    MINIMUM_SCORE = -4
+
+    def self.call(relation: Article.published, options: {})
+      options = DEFAULT_OPTIONS.merge(options)
+      tags, time_ago, count = options.values_at(:tags, :time_ago, :count)
+
+      relation.limit(count)
+      relation = if time_ago == "latest"
+                   relation.order(published_at: :desc).where(score: MINIMUM_SCORE..).presence ||
+                     relation.order(published_at: :desc)
```

Comment:
non blocking to avoid repeating the order clause

```ruby
relation = relation.where(score: MINIMUM_SCORE..).presence || relation
relation.order(published_at: :desc)
```

Reply:
I love this! Thank you for the suggestion. 😊